rocAL/rocAL/source/api/rocal_api_data_loaders.cpp

```diff
@@ -2542,6 +2560,193 @@ rocalRawCIFAR10Source(
     return output;
 }
 
+RocalTensor  ROCAL_API_CALL
+rocalAudioFileSourceSingleShard(
+        RocalContext p_context,
+        const char* source_path,
+        const char* source_file_list_path,
+        unsigned shard_id,
+        unsigned shard_count,
+        bool is_output,
+        bool shuffle,
+        bool loop,
+        float sample_rate,
+        bool downmix,
+        unsigned max_frames,
+        unsigned max_channels,
+        unsigned storage_type,
+        bool stick_to_shard,
+        int shard_size)
+{
+    rocalTensor* output = nullptr;
+    auto context = static_cast<Context*>(p_context);
+    try
+    {
+        if(shard_count < 1 )
+            THROW("Shard count should be bigger than 0")
+
+        if(shard_id >= shard_count)
+            THROW("Shard id should be smaller than shard count")
+        auto [max_frames, max_channels] = evaluate_audio_data_set(StorageType::FILE_SYSTEM, DecoderType::SNDFILE,
+                                                       source_path, "");
+        INFO("Internal buffer size for audio frames = "+ TOSTR(max_frames))
+
+        RocalTensorDataType tensor_data_type = RocalTensorDataType::FP32;
+        unsigned num_of_dims = 3;
+        std::vector<size_t> dims;
+        dims.resize(num_of_dims);
+        dims.at(0) = context->user_batch_size();
+        dims.at(1) = max_frames;
+        dims.at(2) = max_channels;
+        // [bs][sam][c] - 3D
+        // [bs][h][w][c] - [bs][bins][frames]
+        auto info  = rocalTensorInfo(std::vector<size_t>(std::move(dims)),
+                                context->master_graph->mem_type(),
+                                tensor_data_type);
+        info.set_tensor_layout(RocalTensorlayout::NONE);
```

Comment:
If default layout is None we need not set it here

Reply:
Removed